pkg/registry/registry/repository/storage/storage.go

```diff
@@ -174,10 +180,59 @@ func (r *REST) Delete(ctx context.Context, name string, deleteValidation rest.Va
 			return nil, false, err
 		}
 	}
+	repoName := fmt.Sprintf("%s/%s", o.Spec.NamespaceName, o.Spec.Name)
```

Comment:
when you have  distributionClient  you should check r.distributionClient  != nil, then balabala

Reply:
done